tests/integration/security/extra_trust_anchor/extra_trust_anchor_test.go

```diff
@@ -0,0 +1,220 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package extratrustanchor
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	kubeApiCore "k8s.io/api/core/v1"
+	kubeApiMeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/citadel"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/security/pkg/k8s/configmap"
+	"istio.io/istio/security/pkg/k8s/controller"
+	"istio.io/istio/tests/integration/security/util/secret"
+)
+
+func serialNumbersFromCert(cert []byte) []string {
+	var sn []string
+	for len(cert) > 0 {
+		var block *pem.Block
+		block, cert = pem.Decode(cert)
+		if block == nil {
+			break
+		}
+		if block.Type != "CERTIFICATE" || len(block.Headers) != 0 {
+			continue
+		}
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			continue
+		}
+		sn = append(sn, cert.SerialNumber.String())
+	}
+	return sn
+}
+
+func strLess(a, b string) bool { return a < b }
+
+type operation int
+
+const (
+	opAdd operation = iota
+	opRemove
+)
+
+func createConfigMapOrFail(t test.Failer, env *kube.Environment, namespace string, cm *kubeApiCore.ConfigMap) *kubeApiCore.ConfigMap {
+	t.Helper()
+	created, err := env.CreateConfigMap(namespace, cm)
+	if err != nil {
+		t.Fatalf("Could not create configmap %v in namespace %v: %v", cm.Name, namespace, err)
+	}
+	return created
+}
+
+func deleteConfigMapOrFail(t test.Failer, env *kube.Environment, namespace, name string) {
+	t.Helper()
+	if err := env.DeleteConfigMap(namespace, name); err != nil {
+		t.Fatalf("Could not delete configmap %v in namespace %v: %v", name, namespace, err)
+	}
+}
+
+const waitDelay = time.Second
+
+func updateAndCheck(t *testing.T, env *kube.Environment, c citadel.Instance, cm *kubeApiCore.ConfigMap, namespace string, op operation, roots []string) {
```

Comment:
Just want to call out: we check the secret but not the e2e envoy to envoy flow. I imagine that's what being aimed at.